src/main/scala/KindProjector.scala

```diff
@@ -200,19 +200,19 @@ class KindRewriter(plugin: Plugin, val global: Global)
 
       tree match {
         case Template(parents, self, body) =>
-          Template(parents.map(transform), self, body.map(transform))
+          treeCopy.Template(tree, parents.map(transform), self, body.map(transform))
```

Comment:
I'll try to look into it more, and maybe test out disabling it. I was looking at that case earlier today too.